packages/core/src/api/toDecimal.ts

```diff
@@ -32,9 +41,15 @@ export function toDecimal<TAmount, TOutput>(calculator: Calculator<TAmount>) {
     assert(isDecimal, NON_DECIMAL_CURRENCY_MESSAGE);
 
     const units = toUnitsFn(dineroObject);
+    const isNegative = units.some((unit) => lessThanFn(signFn(unit), zero));
```

Comment:
We should be able to simplify this further and not even have to map over units (either here or even reusing the existing `Array.map` in `getDecimal`).

Normally, when using `toUnits` with a decimal Dinero objects (base 10 with a single base) it should always return a tuple of 2.

```js
toUnits(dinero({ amount: -1, currency: USD })) // [-0, -1]
toUnits(dinero({ amount: -0, currency: USD })) // [-0, -0]
```

If the first unit is zero, then we can just test the second unit for being negative instead of mapping over all units. This way, the second test only happens when necessary (when the first unit is zero) and we avoid an unnecessary iteration.

Reply:
Yes, that makes good sense. Can we always assume there's a second value in the tuple? I used `units.length - 1` just in case that's not always the case.

Reply:
When wasn't it the case? With decimal objects it should be.